search/retrieval-augmented-generation/rag_with_dual_llms/src/vertex_rag_demo_dual_llms_with_judge.py

```diff
@@ -0,0 +1,1177 @@
+    except Exception as e:
+        st.error(f"Failed to list Gemini models: {e}")
+        logging.error(f"Gemini model initialization error: {e}")
+
+    # Initialize Ollama models (if server is running)
+    try:
+        response = requests.get(DEFAULT_OLLAMA_URL, timeout=DEFAULT_OLLAMA_TIMEOUT)
+        response.raise_for_status()  # Raises HTTPError for bad responses (4xx or 5xx)
+        ollama_models = [model["name"] for model in response.json().get("models", [])]
+        log_debug(f"Found Ollama models: {ollama_models}")
+        if not ollama_models:
+             st.warning("Ollama server responded, but no models listed.")
+    except requests.exceptions.ConnectionError:
+        st.warning("🔴 Ollama server not reachable. Is it running?")
+        logging.warning("Ollama connection failed: ConnectionError.")
+    except requests.exceptions.Timeout:
+        st.warning("🔴 Ollama server timed out.")
+        logging.warning("Ollama connection failed: Timeout.")
+    except requests.exceptions.RequestException as e:
+        st.error(f"Ollama connection failed: {e}")
+        logging.error(f"Ollama connection failed: {e}")
+    except Exception as e:
+         st.error(f"Error processing Ollama response: {e}")
+         logging.error(f"Error processing Ollama response: {e}")
+
+
+    return ollama_models, gemini_models
+
+
+def generate_gemini_response(
+    prompt: str, model_config: Dict[str, Any], col: st.container
+) -> str:
+    """
+    Generates a response using the Gemini Chat API via google-genai client.
+
+    Args:
+        prompt: The complete prompt string.
+        model_config: Dictionary containing 'name' and 'temperature'.
+        col: The Streamlit column container to display the response in.
+
+    Returns:
+        The generated response text.
+    """
+    start_time = (time.time())*1000.0
+    model_name = model_config["name"]
+    temperature = model_config.get("temperature", DEFAULT_GEMINI_TEMPERATURE)
+    session_key = f"gemini_chat_client_{model_name}_{col}" # More specific key
+
+    log_debug(
+        f"Calling Gemini API: model={model_name}, temp={temperature}"
+    )
+    log_debug(f"Full prompt for Gemini:\n{prompt[:500]}...") # Log truncated prompt
+
+    full_response = ""
+    system_instruction = st.session_state.get("system_prompt", "").strip()
+
+    # Configuration for the generation request
+    # Note: system_instruction placement differs for chat vs. generate_content
+    generate_content_config = types.GenerateContentConfig(
+        temperature=temperature,
+        top_p=DEFAULT_GEMINI_TOP_P,
+        max_output_tokens=DEFAULT_GEMINI_MAX_TOKENS,
+        # response_modalities=["TEXT"], # Often not needed for text-only models
+        system_instruction=[types.Part.from_text(text=system_instruction)],
+    )
+
+    chat_content_config = {
+        "system_instruction":system_instruction,
+        "temperature": temperature,
+        "max_output_tokens": DEFAULT_GEMINI_MAX_TOKENS,
+    }
```

Comment:
Can you change this to use the `GenerateContentConfig` class or whichever class is needed for chat input?

Reply:
Actually for now, the chat config is simply a JSON like this in the new API. If you tried any of the GenerateContentConfig() and pass it to the new chat API, it will give an error. C'est la vie!